dapplet/src/index.ts

```diff
@@ -151,121 +145,214 @@ export default class TwitterFeature {
       me.loading = true;
-      const linkedAccount = await this.identityService.getNearAccount('twitter/' + profile.id);
       me.label = 'Waiting...';
-      if (linkedAccount) {
+      if (walletForAutoclaim) {
         // unlink
-        if (nearAccount) {
-          alert('Remove NEAR Account ID from your profile name before continue.');
-        } else {
-          await this.identityService.requestVerification(
-            `twitter/${profile.id}`,
-            true,
-            'https://twitter.com/' + profile.id,
-          );
-        }
+        await this.tippingService.deleteWalletForAutoclaim(accountGId);
       } else {
         // link
-        if (!nearAccount) {
-          const exampleWallet = this._network === NearNetwork.TESTNET ? 'yourwallet.testnet' : 'yourwallet.near';
+        if (!parsiedNearAccount) {
+          const exampleWallet = this._network === NearNetworks.Testnet ? 'yourwallet.testnet' : 'yourwallet.near';
           alert(
-            'Add your NEAR account ID to your profile name in Twitter before continuing. ' +
+            `Before you continue, add your NEAR account ID to your ${websiteName} profile name. ` +
               'This is necessary for Oracle so that it can make sure that you own this Twitter account. ' +
               'After linking you can remove it back.\n' +
               `For example: "${profile.authorFullname} (${exampleWallet})"\n`,
           );
         } else {
-          await this.identityService.requestVerification(
-            `twitter/${profile.id}`,
-            false,
-            'https://twitter.com/' + profile.id,
-          );
+          const prevSessions = await Core.sessions();
+          const prevSession = prevSessions.find((x) => x.authMethod === 'near/' + this._network);
+          const session = prevSession ?? (await Core.login({ authMethods: ['near/' + this._network] }));
+          const wallet = await session.wallet();
+          if (parsiedNearAccount !== wallet.accountId) {
+            alert(
+              `Check the wallet in your ${websiteName} profile name. ` +
+                'It should be the same as you use for login. Now you logged in with ' +
+                `${wallet.accountId}, and the wallet in your profile is ${parsiedNearAccount}.`,
+            );
+          } else {
+            const args = {
+              firstAccountId: profile.id,
+              firstOriginId: websiteNameLowerCase,
+              firstAccountImage: img,
+              secondAccountId: parsiedNearAccount,
+              secondOriginId: 'near/' + this._network,
+              secondAccountImage: null,
+              isUnlink: false,
+              firstProofUrl: 'https://' + websiteNameLowerCase + '.com/' + profile.id,
+            };
+            const condition = {
+              type: `${websiteNameLowerCase}/near-${this._network}`,
+              user: fullname,
+            };
+            await Core.connectedAccounts.requestVerification(args, condition);
```

Comment:
Can this logic be moved to a separate service to make the dapplet code cleaner?

Reply:
fixed